libhdf5/hdf5create.c

```diff
@@ -157,12 +166,18 @@ nc4_create_file(const char *path, int cmode, size_t initialsz,
 
 #ifdef HAVE_H5PSET_LIBVER_BOUNDS
 #if H5_VERSION_GE(1,10,2)
-    if (H5Pset_libver_bounds(fapl_id, H5F_LIBVER_EARLIEST, H5F_LIBVER_V18) < 0)
+    low = H5F_LIBVER_EARLIEST;
+    high = H5F_LIBVER_V18;
+    if ((cmode & NC_HDF5_SWMR)) {
+      low = H5F_LIBVER_LATEST;
+      high = H5F_LIBVER_LATEST;
```

Comment:
Will this create files that are unreadable by older versions of HDF5?

Reply:
Note that `H5F_LIBVER_V18` was introduced in 1.10.2 so maybe the `HAVE_H5PSET_LIBVER_BOUNDS` is checking for that version and not for existance of `H5Pset_libver_bounds`

Reply:
I think that the files will be unreadable with 1.8.X clients.  Here is a blurb from 1.10.2 release notes:

> When it is used in application linked with HDF5 1.10.0, it will enable new chunk indexing for Single Writer/Multiple Reader (SWMR) access and Virtual Dataset storage. 

>What does this change mean to an HDF5 application?
When an HDF5 application linked with HDF5 1.10.2 specifies H5F_LIBVER_LATEST as a value for the “high” parameter, the application may produce files that are not compatible with the HDF5 1.8.* file format.  For example, new chunk indexing will be used that was not known to HDF5 1.8.*. This means that an application linked with HDF5 1.8.* libraries may not be able to read such files.

>When an HDF5 application linked with HDF5 1.10.2 specifies H5F_LIBVER_V18 as a value for the “high” parameter, the application will produce files fully compatible with HDF5 1.8.*, meaning that any application linked with the HDF5 1.8.* libraries will be able to read such files.